addons/mail/models/discuss/discuss_channel_category.py

```diff
@@ -0,0 +1,30 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+from odoo.addons.mail.tools.discuss import Store
+
+
+class ChannelCategory(models.Model):
+    _name = "discuss.channel.category"
+    _description = "Channel Category"
+
+    name = fields.Char('Channel category name', required=True)
+    channel_ids = fields.One2many('discuss.channel', 'channel_category_id', string='Channels')
+
+    @api.model
+    def get_categories(self):
+        categories = self.env['discuss.channel.category'].search([])
+        return Store(categories).get_result()
+
+    def _to_store(self, store: Store):
+        """Adds channel categories data to the given store."""
+        fields = {
+            "name": True,
+        }
+        for category in self:
+            data = category._read_format(
+                list(fields),
+                load=False,
+            )[0]
+            data["threads"] = Store.many(category.channel_ids)
```

Comment:
We should not do renames in new models, fields name should fully match python/js (with the python syntax)

Reply:
it is not really a new model.

it is a new model python side but it maps to the existing DiscussAppCategory js model, where the list of channels is already called "threads".

Reply:
Better rename the JS field to `channel_ids`, as the python has authority on naming of fields